client/pages/database/roles.ts

```diff
@@ -0,0 +1,30 @@
+export const roles = {
```

Comment:
Could you pull from the `master` branch and then move this `client/pages/database/roles.ts` to `client/database/roles.ts`?

I think we will have a build error if we put our in-memory database under the `pages` folder.